consensus/misc/eip1559.go

```diff
@@ -33,11 +34,23 @@ const MaximumL2BaseFee = 10000000000
 // `scalar` accounts for finalization costs. `overhead` accounts for sequencing and proving costs.
 // we use 1e18 for precision to match the contract implementation.
 var (
-	BaseFeePrecision       = new(big.Int).SetUint64(1e18)
-	DefaultBaseFeeScalar   = new(big.Int).SetUint64(34000000000000)
-	DefaultBaseFeeOverhead = new(big.Int).SetUint64(15680000)
+	BaseFeePrecision = new(big.Int).SetUint64(1e18)
+
+	// scalar and overhead are updated automatically in `Blockchain.writeBlockWithState`.
+	baseFeeScalar   = new(big.Int).SetUint64(34000000000000)
```

Comment:
these parameters are set via `UpdateL2BaseFeeParams` which is called when processing a block that contains the update event. What I don't understand: how is this persisted across node startups? What if a node read and updated these values and shuts down. Now it won't process this block anymore and needs to read from state if these values are at default values no?

Reply:
Added in [9bda21c](https://github.com/scroll-tech/go-ethereum/pull/1189/commits/9bda21c3a3e8d7957acee8e6a30abcab1e356752)